### PR TITLE
Add Johari Window self-awareness visualization

### LINKED ISSUE #282: Johari Window Visualization: Self-Awareness Mapping
## Overview
Implement an interactive Johari Window visualization that maps the delta between a user's self-assessment (from reflection prompts) and AI behavioral analysis to reveal blind spots, track self-awareness growth over time, and guide development focus areas.

## Motivation
The [Johari Window](https://psychsafety.com/the-johari-window/) (Luft & Ingham, 1955) is a foundational framework for interpersonal awareness. It divides self-knowledge into four quadrants based on what is known/unknown to self and others. By comparing what users *think* happened in meetings (self-reflection) with what the AI *detected* (behavioral analysis), we can surface blind spots and track awareness growth over time.

## The Four Quadrants

| | Known to Self | Unknown to Self |
|---|---|---|
| **Known to Others (AI)** | **Open** — Aligned self-awareness | **Blind Spot** — AI sees it, you don't |
| **Unknown to Others** | **Hidden** — Private reflections | **Unknown** — Emerges over time |

### How Each Quadrant Maps to Data

- **Open Area**: Self-assessment matches AI analysis (e.g., user said "too much talk time" and AI confirms >50%)
- **Blind Spot**: AI detects a pattern the user didn't acknowledge (e.g., user said "about right" talk time but AI shows 62%, or user was unaware of interruptions)
- **Hidden Area**: Freeform reflections about internal experiences not visible in transcript (e.g., "I felt anxious about the budget discussion")
- **Unknown**: New patterns that emerge over time from aggregate data that neither the user nor single-meeting analysis caught

### Backend

- [ ] \`GetJohariWindowData\` query:
  - Aggregates reflection vs analysis deltas across meetings
  - Calculates quadrant sizes based on alignment/misalignment rates
  - Returns:
    - Quadrant proportions (Open %, Blind Spot %, Hidden %, Unknown %)
    - Specific blind spot items (with meeting references)
    - Trend data (quadrant proportions over time)
    - Growth indicators (is Open area expanding?)
- [ ] Scoring logic:
  - For each reflected meeting, compare structured self-assessments to analysis data
  - Aligned = Open, Misaligned = Blind Spot
  - Freeform reflections without corresponding analysis data = Hidden
  - Patterns detected only in aggregate = Unknown

### API Layer
- [ ] \`GET /api/insights/johari-window?range=90d\`
- [ ] Response includes quadrant data, blind spot details, and trend

### Frontend

- [ ] Johari Window visualization on the Insights page (alongside #280 trends):
  - 2x2 grid with quadrant sizes proportional to data
  - Color-coded quadrants (e.g., green=Open, amber=Blind Spot, blue=Hidden, gray=Unknown)
  - Click a quadrant to drill into details
- [ ] Blind Spot detail view:
  - List of specific blind spots with meeting references
  - "You rated your talk time as 'about right' in 4 of 6 meetings, but analysis showed >50% in all 4"
  - "You were unaware of interruptions in 3 meetings"
- [ ] Open Area growth tracker:
  - Simple trend line showing Open area % over time
  - "Your self-awareness has improved by 15% over the last 30 days"
- [ ] Guidance text explaining each quadrant in plain language
- [ ] Empty state when insufficient data (need minimum N reflected meetings)
- [ ] Mobile responsive

### Visualization

```
┌─────────────────────────────────────────────────────────────┐
│  Your Johari Window                    Last 90 days         │
├─────────────────────────────────────────────────────────────┤
│                                                             │
│  ┌────────────────────────┬─────────────────┐               │
│  │                        │                 │               │
│  │       OPEN (62%)       │  BLIND SPOT     │               │
│  │                        │    (18%)        │               │
│  │  Known to you          │  Others see it, │               │
│  │  and visible to others │  you don't yet  │               │
│  │                        │                 │               │
│  ├────────────────────────┼─────────────────┤               │
│  │                        │                 │               │
│  │     HIDDEN (15%)       │  UNKNOWN (5%)   │               │
│  │                        │                 │               │
│  │  Your private          │  Yet to be      │               │
│  │  reflections           │  discovered     │               │
│  │                        │                 │               │
│  └────────────────────────┴─────────────────┘               │
│                                                             │
│  📈 Your Open area has grown 8% this month                  │
│                                                             │
│  Top Blind Spots:                                           │
│  • Talk time perception (misaligned in 4/6 meetings)        │
│  • Interruption awareness (unaware in 3 meetings)           │
│  • Tone shifts during budget topics (2 meetings)            │
│                                                             │
└─────────────────────────────────────────────────────────────┘
```

### Minimum Data Thresholds
- Need at least 3 meetings with both analysis AND reflection to show the window
- Below threshold, show progress indicator: "Reflect on 2 more meetings to unlock your Johari Window"

### Quadrant Calculation
- Each reflected meeting produces alignment scores per dimension
- Aggregate alignment rate = Open area proportion
- Misalignment rate = Blind Spot proportion
- Hidden = number of freeform reflections without corresponding metrics
- Unknown = remainder (starts large, shrinks as data grows)

### Growth Tracking
- Store periodic snapshots of quadrant proportions
- Compare current period to previous to calculate growth

## Dependencies
- #281 (Post-Meeting Reflection Prompts) — provides self-assessment data
- #280 (Personal Insights Dashboard) — shares the Insights page

## Feeds Into
- #283 (Behavioral Goals) — blind spots suggest goals to set

## Part of
Self-Improvement Insights Feature Suite

### PR DESCRIPTION
**Pull request by garethbaumgart**
## Summary
- Implements classic 2×2 proportional grid visualization comparing user self-reflections against AI behavioral analysis across 4 dimensions (talk time, engagement, tone, interruptions) to surface awareness gaps
- Classifies each meeting into Open, Blind Spot, Hidden, and Unknown quadrants with trend tracking and detailed blind spot explanations
- Includes full loading skeleton, error state, and empty state with progress dots for meetings needing reflections

Closes #282

### Backend
- `GetJohariWindow` query: fetches meetings with both analysis and reflection data, classifies 4 dimensions per meeting, aggregates to percentages, calculates Open trend over time
- `JohariWindowDto` records: response DTOs for the 2×2 grid, dimension breakdown, and blind spot details
- `GET /api/insights/johari-window?range=` endpoint with range validation
- 30 unit tests covering classification logic, aggregation, trend calculation, date filtering, and edge cases

### Frontend
- `JohariWindowComponent`: 2×2 CSS grid with proportional `fr` sizing (min 5fr floor), dimension breakdown cards, blind spot detail list, growth trend banner, and meeting count footer
- `JohariWindowService`: signals-based HTTP service with loading/error state
- Integrated into insights page (visible in both empty and full states)

## Test plan
- [x] All 587 unit tests pass (412 domain + 175 application including 30 new Johari Window tests)
- [x] All 25 E2E tests pass
- [x] Frontend builds clean
- [ ] Visual verification of 2×2 grid with proportional sizing
- [ ] Empty state displays correctly with progress dots
- [ ] Date range selector reloads Johari Window data
- [ ] Dark mode renders correctly

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Add a Johari Window self-awareness insight that compares user reflections with AI behavioral analysis over recent meetings and exposes it via API and the insights UI.

New Features:
- Introduce a Johari Window insights endpoint that aggregates meeting analysis and reflection data into quadrant percentages, dimension summaries, and blind spot details.
- Add Angular Johari Window component and service to render a proportional 2×2 grid visualization with trend banner, dimension breakdown, blind spot list, and empty/loading/error states on the insights page.

Enhancements:
- Wire the Johari Window use case into the application dependency injection and insights routing with range validation for supported time windows.

Tests:
- Add comprehensive unit tests for Johari Window classification logic, aggregation, date range filtering, trend calculation, and valid range handling.

## Summary by CodeRabbit

* **New Features**
  * Johari Window insight added to Insights page: interactive 2x2 grid, trend banner, legend, per‑dimension breakdown, blind‑spot details, loading/error/empty states and accessibility.
  * Time‑range selection (7d, 30d, 90d, all) with meeting count and progress indication.

* **Backend**
  * New API to serve Johari Window insights and a service that computes aggregated Johari metrics.

* **Tests**
  * Comprehensive tests covering classifications, range filtering, data sufficiency, aggregation and trends.

* **Bug Fixes**
  * Range handling made case‑insensitive.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/PraxisNote.Application/Features/Insights/GetJohariWindow.cs [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+    internal static int CalculateHiddenCount(ReflectionDto reflection)
+    {
+        var count = 0;
+
+        // Freeform reflection represents private self-knowledge not in analysis
+        if (!string.IsNullOrWhiteSpace(reflection.FreeformReflection)
+            && reflection.FreeformReflection.Length > MinFreeformReflectionLength)
+            count++;
+
+        return count;
```
[the remark this answers is not in the document]
Comment (garethbaumgart) [reply src/PraxisNote.Application/Features/Insights/GetJohariWindow.cs]: Not addressing this one. In Johari Window theory, the Hidden quadrant represents things the person knows about themselves but doesn't share with others. A freeform reflection captures exactly that — private self-knowledge not present in the AI behavioral analysis — and this is independent of whether structured self-assessments exist.

A user can simultaneously:
- Have structured fields (talk time, engagement, tone, interruptions) → these classify into Open/BlindSpot/Unknown
- Write a freeform reflection about their internal experience → this is Hidden knowledge

Gating Hidden on the absence of structured fields would mean the Hidden quadrant could only exist when the user filled out freeform text but skipped all structured fields, which is a rare edge case and would effectively eliminate the Hidden quadrant from the model.
